### ISSUE TITLE
[Bug]: Next and previous buttons in cds-pagination are violating a11y rules.

### ISSUE THREAD

**Issue by rsimpso1** [carbon-design-system/carbon#17767]
### Package

@carbon/web-components

### Browser

Chrome

### Package version

2.13

### React version

18.2.0

### Description

The pagination buttons are violating a11y rules. If you click until the button becomes disabled, focus is completely lost.

### Reproduction/example

https://stackblitz.com/edit/github-kbd9xw-s3y3s6?file=index.html

### Steps to reproduce

Click the next or previous button and watch focus be lost. (I'm marking as a sev3 because it isn't that severe, but also to be clear, I don't have a workaround)

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Application/PAL

watsonx Assistant

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
